### DISCUSSION TITLE
Matching on constants in a bitstring case expression

### DISCUSSION THREAD

**Discussion by Xetera** [gleam-lang/gleam#2333] [Questions & support] [answered]
First of all, amazing work! I'm loving everything in the language so far.

One thing that came up while I was trying to work on a jpeg parser was trying to give clear names to bitstring patterns I want to match on. I was able to get something working with

```gleam
fn parse_segment(marker: BitString, bytes: BitString) {
  case marker {
    <<0xFF, 0xE:4, app_number:4>> -> {
      app.parse(bytes, app_number)
    }
    <<0xFF, 0xC:4, identifier:4>> -> {
      frame.parse(bytes, identifier)
    }
  }
}
```

But what I wanted was something more along the lines of

```gleam
const marker_prefix = 0xFF
const app_marker = 0xE
const frame_marker = 0xC

fn parse_segment(marker: BitString, bytes: BitString) {
  case marker {
    <<marker_prefix, app_marker:4, app_number:4>> -> {
      app.parse(bytes, app_number)
    }
    <<marker_prefix, frame_marker:4, identifier:4>> -> {
      frame.parse(bytes, identifier)
    }
  }
}
```

but when I do that, the `marker_prefix` and `app_marker` variables matched against the bitstring are interpreted as a pattern binding for a new variable instead. Is there any syntax to prevent repetition here? I couldn't find anything in the documentation about this.

In elixir the ambiguity between the two isn't an issue when using module attributes so maybe this isn't such a simple problem to solve

**Comment by lpil** [comment, marked as answer]
Hello! In Gleam we don't have one syntax for both equality and also for assigning variables, so instead guards are used.

```gleam
<<m, a:4, f:4>> if m == marker_prefix && a == app_marker && f == frame_marker -> todo
```

**Reply by Xetera** [reply]
Thanks, that'll work just fine 👍
